STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/UsersService.java

```diff
@@ -75,6 +77,7 @@
     private final PropertyResolver environment;
     private final PasswordEncoder passwordEncoder;
     private final ExternalGroupService extGroupService;
+    private final MailService mailService;
```

Comment:
Please, DON'T use RS WS service in another service class. Basically, it's a controller class and must be used only for user communication purposes. If this logic is supposed to be used in another place, it means that this logic must be moved to the separated bean.
An application must be designed in the following way:

- Web Layer (Controllers, exception handlers, filters, RS WS Services and etc.)
- Service Layer (Classes with business logic)
- DAO (DB Repositories)